src/crs/projected/eqareaalbers.jl

```diff
@@ -0,0 +1,155 @@
+Albers(1.0km, 1.0km) # length quantities are converted to meters
+Albers(1.0m, 1.0m)
+Albers{NAD83}(1.0m, 1.0m)
+```
+
+See [EPSG:5070](https://epsg.io/5070).
+"""
+struct Albers{latₒ,lat₁,lat₂,Datum,Shift,M<:Met} <: Projected{Datum,Shift}
+  x::M
+  y::M
+end
+
+Albers{latₒ,lat₁,lat₂,Datum,Shift}(x::M, y::M) where {latₒ,lat₁,lat₂,Datum,Shift,M<:Met} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift,float(M)}(x, y)
+Albers{latₒ,lat₁,lat₂,Datum,Shift}(x::Met, y::Met) where {latₒ,lat₁,lat₂,Datum,Shift} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift}(promote(x, y)...)
+Albers{latₒ,lat₁,lat₂,Datum,Shift}(x::Len, y::Len) where {latₒ,lat₁,lat₂,Datum,Shift} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift}(uconvert(m, x), uconvert(m, y))
+Albers{latₒ,lat₁,lat₂,Datum,Shift}(x::Number, y::Number) where {latₒ,lat₁,lat₂,Datum,Shift} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift}(addunit(x, m), addunit(y, m))
+
+Albers{latₒ,lat₁,lat₂,Datum}(args...) where {latₒ,lat₁,lat₂,Datum} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift()}(args...)
+
+Albers(args...) = Albers{NAD83}(args...)
+
+Base.convert(
+  ::Type{Albers{latₒ,lat₁,lat₂,Datum,Shift,M}},
+  coords::Albers{latₒ,lat₁,lat₂,Datum,Shift}
+) where {latₒ,lat₁,lat₂,Datum,Shift,M} = Albers{latₒ,lat₁,lat₂,Datum,Shift,M}(coords.x, coords.y)
+
+constructor(::Type{<:Albers{latₒ,lat₁,lat₂,Datum,Shift}}) where {latₒ,lat₁,lat₂,Datum,Shift} =
+  Albers{latₒ,lat₁,lat₂,Datum,Shift}
+
+lentype(::Type{<:Albers{latₒ,lat₁,lat₂,Datum,Shift,M}}) where {latₒ,lat₁,lat₂,Datum,Shift,M} = M
+
+==(
+  coords₁::Albers{latₒ,lat₁,lat₂,Datum,Shift},
+  coords₂::Albers{latₒ,lat₁,lat₂,Datum,Shift}
+) where {latₒ,lat₁,lat₂,Datum,Shift} = coords₁.x == coords₂.x && coords₁.y == coords₂.y
+
+# ------------
+# CONVERSIONS
+# ------------
+
+# Adapted from PROJ coordinate transformation software
+# Initial PROJ 4.3 public domain code was put as Frank Warmerdam as copyright
+# holder, but he didn't mean to imply he did the work. Essentially all work was
+# done by Gerald Evenden.
+# Authors of the original algorithm: Gerald Evenden and Thomas Knudsen
+# reference code: https://github.com/OSGeo/PROJ/blob/master/src/projections/aea.cpp
+
+inbounds(::Type{<:Albers}, λ, ϕ) =
+  -π ≤ λ ≤ π && deg2rad(90) ≤ ϕ ≤ deg2rad(90)
+
+function formulas(::Type{<:Albers{latₒ,lat₁,lat₂,Datum}}, ::Type{T}) where {latₒ,lat₁,lat₂,Datum,T}
+  # Constants
+  🌎 = ellipsoid(Datum)
+  e = T(eccentricity(🌎))
+  a = numconvert(T, majoraxis(🌎))
+
+  # Latitude origin
+  ϕₒ = T(ustrip(deg2rad(latₒ)))
+
+  # Standard parallels
+  ϕ₁ = T(ustrip(deg2rad(lat₁)))
+  ϕ₂ = T(ustrip(deg2rad(lat₂)))
+
+  m₁ = hm(ϕ₁, e)
+  m₂ = hm(ϕ₂, e)
+  α₁ = hα(ϕ₁, e)
+  α₂ = hα(ϕ₂, e)
+  αₒ = hα(ϕₒ, e)
+  n = (m₁^2 - m₂^2) / (α₂ - α₁)
+  C = m₁^2 + n * α₁
+
+  Θ = n * λ
+  ρ = a * sqrt(C - n * hα(ϕ, e)) / n
+  if ρ < 0
+    throw(ArgumentError("coordinates outside of the projection domain"))
+  end
+  ρₒ = sqrt(a * (C - n * αₒ)) / n
+  function fx(λ, ϕ)
+    ρₒ - ρ * cos(Θ)
+  end
+
+  function fy(λ, ϕ)
+    ρ * sin(Θ)
+  end
+
+  fx, fy
```

Comment:
This implementation needs some fixes:
```suggestion
  🌎 = ellipsoid(Datum)
  e = T(eccentricity(🌎))
  ϕₒ = T(ustrip(deg2rad(latₒ)))
  ϕ₁ = T(ustrip(deg2rad(lat₁)))
  ϕ₂ = T(ustrip(deg2rad(lat₂)))

  m₁ = hm(ϕ₁, e)
  m₂ = hm(ϕ₂, e)
  α₁ = hα(ϕ₁, e)
  α₂ = hα(ϕ₂, e)
  n = (m₁^2 - m₂^2) / (α₂ - α₁)
  C = m₁^2 + n * α₁

  Θ(λ) = n * λ
  ρ(ϕ) = sqrt(C - n * hα(ϕ, e)) / n
  ρₒ = ρ(ϕₒ)

  fx(λ, ϕ) = ρ(ϕ) * sin(Θ(λ))

  fy(λ, ϕ) = ρₒ - ρ(ϕ) * cos(Θ(λ))

  fx, fy
```
First, multiplying the result by `a` (major axis) is done in a post-process step, not here.
Second, the "longitude" (lambda) and "latitude" (phi) arguments are only available inside the `fx` and `fy` functions, only the constants should be left outside these functions, and any value that depends of these arguments should be calculated inside these functions.

Reply:
ooohhh okay. I wondered how to handle that case of functions depending on \lambda and \phi  that are then fed into the final `fx()` and `fy()` functions. good to know that is how to do it.